docs/src/adr/015adr-trace.md

```diff
@@ -0,0 +1,403 @@
+ `p[1]` is the map value. Importantly, a key may be an arbitrary expression. It
+ does not have to be a string or an integer. TLA+ functions are written as maps
+ in this format.
+
+
+### Example
+
+The counterexample to `NoSolution` may be written in the ITF format as follows:
+
+```js
+{
+  "#meta": {
+    "source": "MC_MissionariesAndCannibalsTyped.tla"
+  },
+  "vars": [ "bank_of_boat", "who_is_on_bank" ],
+  "states": [
+    {
+      "#meta": { "index": 0 },
+      "bank_of_boat": "E",
+      "who_is_on_bank": {
+        "#map": [
+          [ "E", { "#set": [ "c1_OF_PERSON", "c2_OF_PERSON",
+                             "m1_OF_PERSON", "m2_OF_PERSON" ] } ],
+          [ "W", { "#set": [] } ]
+        ]  
+      }
+    },
+    {
+      "#meta": { "index": 1 },
+      "bank_of_boat": "W",
+      "who_is_on_bank": {
+        "#map": [
+          [ "E", { "#set": [ "c1_OF_PERSON", "m1_OF_PERSON" ] } ],
+          [ "W", { "#set": [ "c2_OF_PERSON", "m2_OF_PERSON" ] } ]
+        ]  
+      }
+    },
+    {
+      "#meta": { "index": 2 },
+      "bank_of_boat": "E",
+      "who_is_on_bank": {
+        "#map": [
+          [ "E", { "#set": [ "c1_OF_PERSON",
+                             "m1_OF_PERSON", "m2_OF_PERSON" ] } ],
+          [ "W", { "#set": [ "c2_OF_PERSON" ] } ]
+        ]  
+      }
+    },
+    {
+      "#meta": { "index": 3 },
+      "bank_of_boat": "W",
+      "who_is_on_bank": {
+        "#map": [
+          [ "E", { "#set": [ "c1_OF_PERSON" ] } ],
+          [ "W", { "#set": [ "c2_OF_PERSON", "m1_OF_PERSON", "m2_OF_PERSON" ] } ]
+        ]  
+      }
+    },
+    {
+      "#meta": { "index": 4 },
+      "bank_of_boat": "E",
+      "who_is_on_bank": {
+        "#map": [
+          [ "E", { "#set": [ "c1_OF_PERSON", "c2_OF_PERSON" ] } ],
+          [ "W", { "#set": [ "m1_OF_PERSON", "m2_OF_PERSON" ] } ]
+        ]  
+      }
+    },
+    {
+      "#meta": { "index": 5 },
+      "bank_of_boat": "W",
+      "who_is_on_bank": {
+        "#map": [
+          [ "E", { "#set": [ ] } ],
+          [ "W", { "#set": [ "c1_OF_PERSON", "c2_OF_PERSON",
+                             "m1_OF_PERSON", "m2_OF_PERSON" ] } ]
+        ]  
+      }
+    }
+  ]
+}
+```
+
+Compare the above trace format with the TLA+ counterexample. The TLA+ example
+looks more compact. The ITF example is heavier on the brackets and braces, but it is also designed with machine-readability and tool automation in mind, whereas TLA+ counterexamples are not. 
+However, the example in the ITF format is also self-explanatory and does not
+require any understanding of TLA+.
+
+
+## Consequences
+
+<!-- Records the results of the decision over the long term.
+     Did it work, not work, was changed, upgraded, etc.
+-->
+
+Nothing yet. However, we expect this format to become an exchange format
+for the tools that consume traces at their input:
+
+ - End-to-end testing tools,
+
+ - Bug reproduction tools,
+
+ - Trace visualization tools,
+
+ - Trace fuzzers.
```

Comment:
This is space is for retrospection. And we can't retrospect at the outset! I suggest including list useful list in the intro instead, since this is bout motivation, not retrospective evaluation.

```suggestion
```